core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2575,6 +2578,13 @@ private List<AcquiredRecords> filterAbortedTransactionalAcquiredRecords(
         return filterRecordBatchesFromAcquiredRecords(acquiredRecords, recordsToArchive);
     }
 
+    private void logError(String message, Throwable e) {
+        log.debug(message, e);
+        if (!(e instanceof NetworkException)) {
+            log.error(message, e);
+        }
```

Comment:
If not NetworkException then same error message will be twice in debug mode:
```suggestion
        if (e instanceof NetworkException) {
            log.debug(message, e);
        } else {
             log.error(message, e);
        }
```